app/feature/feature-forever/src/main/kotlin/com/hedvig/android/feature/forever/ui/DiscountPieChart.kt

```diff
@@ -0,0 +1,100 @@
+package com.hedvig.android.feature.forever.ui
+
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.InfiniteRepeatableSpec
+import androidx.compose.animation.core.KeyframesSpec
+import androidx.compose.animation.core.RepeatMode
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.material3.typeElement
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+
+@Composable
+fun ColumnScope.DiscountPieChart(
+  containerColor: Color = MaterialTheme.colorScheme.typeElement,
+  discountColor: Color = MaterialTheme.colorScheme.secondaryContainer,
+  incentiveColor: Color = MaterialTheme.colorScheme.surface,
+  totalPrice: Float,
+  totalDiscount: Float,
+  incentive: Float,
+) {
+  val transition = rememberInfiniteTransition(label = "transition")
+  val sweep by transition.animateFloat(
+    initialValue = 0f,
+    targetValue = calculateAngle(totalPrice, incentive),
+    animationSpec = InfiniteRepeatableSpec(
+      animation = KeyframesSpec(
+        KeyframesSpec.KeyframesSpecConfig<Float>().apply {
+          0f at 800 with FastOutSlowInEasing
+          30f at 1300
+          30f at 2800 with FastOutSlowInEasing
+          durationMillis = 2800
+        },
+      ),
+      repeatMode = RepeatMode.Reverse,
```

Comment:
I feel like maybe we don't need to have the animation reverse here but just have it be `RepeatMode.Restart`.
that way we can specify the animation completely in one shot using keyframes again perhaps, and then let it replay from the start again.

something like
```
animation = keyframes {
  durationMillis = 5400
  0f at 0
  0f at 1800
  30f at 2300 with SomeEasingHere
  30f at 3600
  0f at 4100 with SomeEasingHere
  0f at 5400
},
```
So the animations that play from 1800 -> 2300 get the easing, same with 3600 -> 4100

Reply:
The easings actually have to be on `0f at 1800` and `30f at 3600` to have an effect here. But this looks much better, thanks!